cluster-autoscaler/utils/drain/drain_test.go

```diff
@@ -624,7 +645,52 @@ func TestDrain(t *testing.T) {
 		},
 	}
 
-	for _, test := range tests {
+	allTests := []testOpts{}
+	// Note: be careful about modifying the underlying reference values for sharedTest
+	// since they are shared (changing it once will change it for all shallow copies of sharedTest)
+	for _, sharedTest := range sharedTests {
+		// make sure you shallow copy the test like this
+		// before you modify it
+		// (so that modifying one test doesn't affect another)
+		enabledTest := sharedTest
```

Comment:
Note that I am creating a shallow copy of the `sharedTest` I want to modify `sharedTest` once doesn't affect the second modification. e.g.,
```go
for _, sharedTest := range sharedTests {
	sharedTest.skipNodesWithCustomControllerPods = true
	sharedTest.description = fmt.Sprintf("%s with skipNodesWithCustomControllerPods: %v", 
		sharedTest.description, sharedTest.skipNodesWithCustomControllerPods)
	allTests = append(allTests, sharedTest)

	sharedTest.skipNodesWithCustomControllerPods = false
	sharedTest.description = fmt.Sprintf("%s with skipNodesWithCustomControllerPods: %v", 
		sharedTest.description, sharedTest.skipNodesWithCustomControllerPods)
	allTests = append(allTests, sharedTest)
}

fmt.Println("allTests[0]", allTests[0])
fmt.Println("allTests[1]", allTests[1])
```
Prints
```
allTests[0] {RC-managed pod with skipNodesWithCustomControllerPods:true [0xc00018ad80] [] [0xc000498c60] [] false [0xc00018ad80] [] <nil> true}
allTests[1] {RC-managed pod with skipNodesWithCustomControllerPods:true with skipNodesWithCustomControllerPods:false [0xc00018ad80] [] [0xc000498c60] [] false [0xc00018ad80] [] <nil> false}
```
Note the description and the last bool (which is our flag)

To overcome this, I use shallow copy.

Reply:
Output when we use shallow copy
```
allTests[0] {RC-managed pod with skipNodesWithCustomControllerPods:true [0xc000202d80] [] [0xc0004b54a0] [] false [0xc000202d80] [] <nil> true}
allTests[1] {RC-managed pod with skipNodesWithCustomControllerPods:false [0xc000202d80] [] [0xc0004b54a0] [] false [0xc000202d80] [] <nil> false}
```